code/modules/mob/living/login.dm

```diff
@@ -5,6 +5,10 @@
 	update_stat("mob login")
 	update_sight()
 
+	if(loc)
+		var/turf/T = get_turf(src)
+		update_z(T.z)
+
```

Comment:
Make sure that all other `Login`'s that are children of this call parent, or this will break.

Reply:
Double checked all login call parents

Reply:
may be safer to do:

```
	var/turf/T = get_turf(src)
	if (isturf(T))
		update_z(T.z)
```